metadata_service/dao/publication.py

```diff
@@ -69,14 +74,17 @@ async def add_publication(data: Dict) -> Dict:
       The added Publication object
 
     """
-    collection = await get_collection(COLLECTION_NAME)
-    publication_id = data["id"]
-    await collection.insert_one(data)
+    db_connect = DBConnect()
+    collection = await db_connect.get_collection(COLLECTION_NAME)
+    publication_id = db_connect.get_next_id(COLLECTION_NAME, PREFIX)
+    data["id"] = publication_id
```

Comment:
Cannot mutate a Pydantic model object. Leads to an exception.

Reply:
changed it to data.id